### PR TITLE
Fix:[CI-8877]:Change UI-error for customer

### PR DESCRIPTION
**Pull request by Aishwarya-Lad**
## Describe your changes
Changed UI error message as an ask for 
https://harness.atlassian.net/browse/CI-8877

## Checklist
- [x] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

### Rebase Instructions:
To rebase your branch onto the latest changes in the target branch, simply use the following command in a comment: `/rebase`

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/55761)

### CODE REVIEW

**Comment by Aishwarya-Lad** [comment]
trigger uts

**Comment by Aishwarya-Lad** [comment]
trigger gobuild

**Comment by Aishwarya-Lad** [comment]
trigger uts

**Review by anurag-harness — APPROVED** [review]

File: product/ci/engine/new/executor/addon_executor.go [hunk 1]
```diff
@@ -59,7 +60,7 @@ func ExecuteStepOnAddon(ctx context.Context, step *pb.UnitStep, tmpFilePath stri
 	if err != nil {
 		log.Errorw("Execute step RPC failed", "step_id", stepID,
 			"elapsed_time_ms", utils.TimeSince(st), zap.Error(err))
-		return nil, nil, err
+		return nil, nil, fmt.Errorf("Could not connect to addon client after max retries %v", maxAddonRetries)
```
Comment (anurag-harness) [inline product/ci/engine/new/executor/addon_executor.go:63]: Why did you use fmt.Errorf and not errors.Wrap as used above?

Comment (anurag-harness) [inline product/ci/engine/new/executor/addon_executor.go:63]: Why did you use fmt.Errorf and not errors.Wrap as used above?

**Review by anurag-harness — CHANGES_REQUESTED** [review]

File: product/ci/engine/new/executor/addon_executor.go (hunk 1 above)
Comment (Aishwarya-Lad) [reply product/ci/engine/new/executor/addon_executor.go:63]: errors.Wrap does not support addition of "maxAddonRetries" in the message. Only fmt.Errorf allows for such a format.

**Comment by Aishwarya-Lad** [comment]
trigger gobuild

**Comment by anurag-harness** [comment]
trigger gobuild

File: product/ci/engine/new/executor/addon_executor.go (hunk 1 above)
Comment (rutvijmehta-harness) [inline product/ci/engine/new/executor/addon_executor.go:63]: The Execute RPC can fail with any reason (`exit status 1`, some test failed, etc). We cannot return `Could not connect to addon client`.

You'll have to do a string match where if it's a connection error with addon, you return the above error.

**Review by harshjain12 — APPROVED** [review]

**Pull request closed without merging**